pkg/engine/artifacts.go

```diff
@@ -30,24 +30,25 @@ func kubernetesContainerAddonSettingsInit(p *api.Properties) map[string]kubernet
 	}
 	o := p.OrchestratorProfile
 	k := o.KubernetesConfig
+	// TODO validate that each of these addons are actually wired in to the conveniences in getAddonFuncMap
 	return map[string]kubernetesComponentFileSpec{
 		DefaultHeapsterAddonName: {
 			sourceFile:      "kubernetesmasteraddons-heapster-deployment.yaml",
-			base64Data:      k.GetAddonScript(HeapsterAddonName),
+			base64Data:      k.GetAddonScript(DefaultHeapsterAddonName),
 			destinationFile: "kube-heapster-deployment.yaml",
-			isEnabled:       !common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.13.0"),
+			isEnabled:       k.IsAddonEnabled(DefaultHeapsterAddonName),
```

Comment:
we could probably get rid of the "default" part for consistency since the other constants don't have the default prefix